pkg/nats/proxy/compute_handler.go

```diff
@@ -59,32 +67,33 @@ func handleRequest(msg *nats.Msg, handler *ComputeHandler) {
 
 	switch method {
 	case AskForBid:
-		processAndRespond(ctx, msg, handler.computeEndpoint.AskForBid)
+		processAndRespond(ctx, handler.conn, msg, handler.computeEndpoint.AskForBid)
 	case BidAccepted:
-		processAndRespond(ctx, msg, handler.computeEndpoint.BidAccepted)
+		processAndRespond(ctx, handler.conn, msg, handler.computeEndpoint.BidAccepted)
 	case BidRejected:
-		processAndRespond(ctx, msg, handler.computeEndpoint.BidRejected)
+		processAndRespond(ctx, handler.conn, msg, handler.computeEndpoint.BidRejected)
 	case CancelExecution:
-		processAndRespond(ctx, msg, handler.computeEndpoint.CancelExecution)
+		processAndRespond(ctx, handler.conn, msg, handler.computeEndpoint.CancelExecution)
 	case ExecutionLogs:
-		processAndRespond(ctx, msg, handler.computeEndpoint.ExecutionLogs)
+		processAndStream(ctx, handler.streamingClient, msg, handler.computeEndpoint.ExecutionLogs)
```

Comment:
Unless we're expected branches that don't call processAndRespond() perhaps it might be easier to read if each branch just set the handler function (with ExecutionLogs also overriding the connection) and make a single call after the switch?

Reply:
That would be nice, though not sure how it will work with ExecutionLogs. They call a different method `processAndStream`, which accepts a streaming client instead of a conn. What is even worse is go generics! You can define a variable to hold the different handlers as each expects a different Request and Response type

Reply:
Yeah it may not be worth it in the end if there's still a special case for ExecutionLogs.